benchexec/tools/symbiotic.py

```diff
@@ -25,19 +25,21 @@ class Tool(OldSymbiotic):
 
     REQUIRED_PATHS_7_0_0 = ["bin", "include", "properties", "lib", "llvm-8.0.1"]
 
-    def executable(self):
+    def executable(self, tool_locator):
         """
         Find the path to the executable file that will get executed.
         This method always needs to be overridden,
         and most implementations will look similar to this one.
         The path returned should be relative to the current directory.
         """
-        exe = util.find_executable("bin/symbiotic", exitOnError=False)
-        if exe:
-            return exe
-        else:
+        # This is used in the function _version_newer_than.
+        # I don't know a better way to do it.
+        self.tool_locator = tool_locator
```

Comment:
`tool_locator` is probably better to not be stored, there is nothing in the API documentation that guarantees that this would work as expected. Furthermore, searching for the tool on disk for every call to `determine_result` is inefficient and not robust (disk content could change).

But the documentation of `BaseTool2` provides a suggestion what to do in this case already.

Reply:
fixed in fb5a18e4382326ea40bdaa4ebfb99b456f6a0ebb